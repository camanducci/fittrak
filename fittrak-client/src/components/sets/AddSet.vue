<template>
  <div>
    <form>
      <input v-model="repetitions" type="text" placeholder="Repetitions" />
      <input v-model="weight" type="text" placeholder="Weight" />
      <div>
        Units:
        <input v-model="unit" type="radio" id="kg" name="unit" value="KG" />
        <label for="kg">KGs</label>
        <input v-model="unit" type="radio" id="lb" name="unit" checked value="LB" />
        <label for="lb">LBs</label>
      </div>
      <button v-on:click.prevent="addSet">Add Set</Button>
    </form>
  </div>
</template>

<script>
import ADD_SET from "@/graphql/mutations/addSet.graphql";

export default {
  name: "AddSet",

  data() {
    return {
      repetitions: "",
      weight: "",
      unit: ""
    };
  },

  methods: {
    addSet() {
      const { exercise } = this.$props;
      const { repetitions, weight, unit } = this;

      if (!(repetitions && weight && unit)) return;

      this.$apollo
        .mutate({
          mutation: ADD_SET,

          variables: {
            exerciseId: exercise.id,
            repetitions,
            weight,
            unit
          }
        })
        .then(() => {
          // Clear out for next entry
          this.repetitions = "";
          this.weight = "";
        });
    }
  },

  props: {
    exercise: Object
  }
};
</script>

<style scoped>
form {
  padding: 5px;
}
input {
  margin: 5px;
}
</style>
