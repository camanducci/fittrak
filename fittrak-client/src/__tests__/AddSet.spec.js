import { shallowMount } from "@vue/test-utils";

import AddSet from "../components/sets/AddSet";

describe("AddSet.vue", () => {
  test("AddSet renders", () => {
    const component = shallowMount(AddSet);
    expect(component.html()).toMatchSnapshot();
  });
});
