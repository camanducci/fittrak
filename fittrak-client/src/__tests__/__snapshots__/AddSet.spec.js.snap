// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSet.vue AddSet renders 1`] = `
<vform-stub>
  <vtextfield-stub placeholder="Repetitions" type="number"></vtextfield-stub>
  <vtextfield-stub placeholder="Weight" type="number"></vtextfield-stub>
  <vradiogroup-stub column="true" height="auto" mandatory="true" row="true" value="LB">
    <vradio-stub color="primary" label="LB" onicon="$vuetify.icons.radioOn" officon="$vuetify.icons.radioOff" value="LB"></vradio-stub>
    <vradio-stub color="primary" label="KG" onicon="$vuetify.icons.radioOn" officon="$vuetify.icons.radioOff" value="KG"></vradio-stub>
  </vradiogroup-stub>
  <vbtn-stub activeclass="v-btn--active" depressed="true" ripple="true" tag="button" type="button" color="secondary">Add Set</vbtn-stub>
</vform-stub>
`;
